### PR TITLE
Added tests for cos_table

### PR DESCRIPTION
**Pull request by nishnik**
Should there be tests for `tan`, `sec`, `cot` etc. ?
But I think that would be redundant.

### CODE REVIEW

File: symengine/tests/basic/test_functions.cpp [hunk 1]
```diff
@@ -1123,6 +1123,64 @@ TEST_CASE("Sin table: functions", "[functions]")
...
+    RCP<const Basic> sq3 = sqrt(i3);
+    RCP<const Basic> sq2 = sqrt(i2);
+
+    // cos(2pi + pi/6) = sqrt(3)/2
+    r1 = cos(add(mul(pi, i2), mul(div(pi, i12), i2)));
+    r2 = div(sq3, i2);
+    REQUIRE(eq(*r1, *r2));
+
+    // cos(n*pi + pi/6) = sqrt(3)/2
+    r1 = sin(add(mul(pi, integer(10)), mul(div(pi, i12), i2)));
```
Comment (certik) [inline symengine/tests/basic/test_functions.cpp]: Did you forget to change `sin` to `cos` here?

**Comment by certik** [comment]
I think we should write tests for `tan`, `cot`, `csc` and `sec`, since they all have code that uses it. The tests are to ensure that the implementation in these functions is correct, even though they all use the same underlying `sin_table`.

File: symengine/tests/basic/test_functions.cpp [hunk 2]
```diff
@@ -1123,6 +1123,64 @@ TEST_CASE("Sin table: functions", "[functions]")
...
+    RCP<const Basic> r2;
+
+    RCP<const Basic> i2 = integer(2);
+    RCP<const Basic> i3 = integer(3);
+    RCP<const Basic> i12 = integer(12);
+    RCP<const Basic> i13 = integer(13);
+    RCP<const Basic> im1 = integer(-1);
+
+    RCP<const Symbol> x = symbol("x");
+    RCP<const Symbol> y = symbol("y");
```
Comment (thilinarmtb) [inline symengine/tests/basic/test_functions.cpp]: Looks like `x` and `y` are not used in the tests. You can remove these lines if not necessary.

Comment (thilinarmtb) [inline symengine/tests/basic/test_functions.cpp]: Update the `Sin` test case as well.

File: symengine/tests/basic/test_functions.cpp [hunk 3]
```diff
@@ -1123,6 +1123,64 @@ TEST_CASE("Sin table: functions", "[functions]")
...
+    // cos(5*pi/12) = (sqrt(3) - 1)/(2*sqrt(2))
+    r1 = cos(mul(div(pi, i12), integer(5)));
+    r2 = div(sub(sq3, one), mul(i2, sq2));
+    REQUIRE(eq(*r1, *r2));
+
+    // cos(pi/12) = (sqrt(3) + 1)/(2*sqrt(2))
+    r1 = cos(div(pi, i12));
+    r2 = div(add(sq3, one), mul(i2, sq2));
+    REQUIRE(eq(*r1, *r2));
+
```
Comment (thilinarmtb) [inline symengine/tests/basic/test_functions.cpp]: Remove the empty line here. Also, do the same with the `Sin` test case.

File: symengine/tests/basic/test_functions.cpp (hunk 1 above)
Comment (CodeMaxx) [inline symengine/tests/basic/test_functions.cpp]: Yes this should be `cos` . Thats why Travis test fails.

Comment (nishnik) [inline symengine/tests/basic/test_functions.cpp]: Sorry! My bad.
Amended it.

**Comment by thilinarmtb** [comment]
Looks good to me.

**Comment by nishnik** [comment]
Hey! How to rename the branch ?

**Comment by thilinarmtb** [comment]
See [here](http://stackoverflow.com/questions/20007578/renaming-a-branch-while-on-pull-request). I guess you will have to create a new PR if you do so. Don't worry about the branch name but rename the PR appropriately.

**Comment by nishnik** [comment]
Why the Travis build failed? 
```
...
[ 17%] Building CXX object symengine/CMakeFiles/symengine.dir/complex.cpp.o

[ 19%] Building CXX object symengine/CMakeFiles/symengine.dir/constants.cpp.o

[ 20%] Building CXX object symengine/CMakeFiles/symengine.dir/add.cpp.o

[ 21%] Building CXX object symengine/CMakeFiles/symengine.dir/mul.cpp.o

[ 23%] Building CXX object symengine/CMakeFiles/symengine.dir/pow.cpp.o

[ 24%] Building CXX object symengine/CMakeFiles/symengine.dir/functions.cpp.o

No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.

The build has been terminated
```

**Comment by certik** [comment]
For the record: superseded by #803.

**Pull request closed without merging**